includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -847,6 +847,12 @@ public function update_order_status() {
 						$intent_id
 					);
 					$order->add_order_note( $note );
+
+					// The order is successful, so update it to reflect that.
+					$order->update_meta_data( '_charge_id', $intent->get_charge_id() );
+					$order->update_meta_data( '_intention_status', $status );
+					$order->save();
```

Comment:
Thank you for reviewing, @marcinbot, I really appreciate it 🙏 

I 💯 agree that this whole file could use a whole lot of refactoring work. A while back I wrote up a bunch of tests to help make that easier to do.

Can I disagree a bit here, though? I think DRYing this code might be too early at this point. The two functions that are most similar are this one and process_payment a little bit above. I think we could do some refactoring there with the switch statements to start. I think putting these three lines into a function may hurt those bigger refactoring efforts. Right now these functions are large, but fairly easy to understand since no jumping around the code is needed.

After writing tests for these functions I ended up reading [99 Bottles](https://www.sandimetz.com/99bottles) and I had a refactoring/OOP epiphany. That's where I'm coming from on this.

What do you think?